internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/ControllerClient.java

```diff
@@ -913,6 +915,17 @@ public MultiNodeResponse listStorageNodes() {
     return request(ControllerRoute.LIST_NODES, newParams(), MultiNodeResponse.class);
   }
 
+  public StoppableNodeStatusResponse getAggregatedHealthStatus(
+      String clusterName,
+      List<String> instances,
+      List<String> toBeStoppedInstances) {
+    QueryParams params = newParams().add(CLUSTER_ID, clusterName)
+        .add(INSTANCES, String.join(",", instances))
+        .add(TO_BE_STOPPED_INSTANCES, String.join(",", toBeStoppedInstances));
```

Comment:
Should use `putStringSet` instead of joining the collection with `,`.

Also, you'd ideally want to take `Set` as the input parameter to the API in the first place as you don't expect the `instances` to be duplicated nor `to_be_stopped_instance` to be duplicated.

Reply:
`instances` needs to be a list as the decision of which nodes are removable depends on that ordering. I don't have any preference of set vs list for `to_be_stopped_instances`

Reply:
Thanks for the context on ordering. The motivation was to leverage the existing collection APIs under `QueryParams`. If order does matter, if we decide to go down the `QueryParams`, would be good to add support for `putStringList`.

Although, from my discussion this afternoon, to truly support the POST request (current helix implementation), we may have to use the `byte[]` payload to put in these details which would render the above comment obsolete.